### ISSUE TITLE
Issue with implementing ByteTracker

### ISSUE THREAD

**Issue by ElNoSabe322** [mikel-brostrom/yolo_tracking#989]
### Search before asking

- [X] I have searched the Yolo Tracking [issues](https://github.com/mikel-brostrom/yolo_tracking/issues) and found no similar bug report.

### Question

Hello @mikel-brostrom. I am now facing another issue when trying to integrate the ByteTrack tracking algorithm. The code snippet I have used to initialize the ByteTracker object is as follows:
```python
# Define the type and parameters of the tracker used
def load_tracker():
    # Define the tracker object
    tracker = BYTETracker(
        track_thresh=0.45,
        match_thresh=0.8,
        track_buffer=30,
        frame_rate=30,
    )
    return tracker

# Initializing the tracker
tracker = load_tracker()
# Update the tracker
tracker_objects = tracker.update(numpy_detections, frame) 
``` 
```numpy_detections``` is a numpy array containing the result of the YOLOv8 detections. 

And this is the error I receive: 
```
0: 512x640 2 Turtles, 176.6ms
Speed: 4.0ms preprocess, 176.6ms inference, 9.1ms postprocess per image at shape (1, 3, 512, 640)

0: 512x640 2 Turtles, 109.7ms
Speed: 4.0ms preprocess, 109.7ms inference, 3.0ms postprocess per image at shape (1, 3, 512, 640)
Traceback (most recent call last):
  File "c:\Users\Delta\YOLOv8_GUtech_Turtle\Turtle_Detection\Detect_Track.py", line 565, in <module>
    main()
  File "c:\Users\Delta\YOLOv8_GUtech_Turtle\Turtle_Detection\Detect_Track.py", line 541, in main
    videoTrack()
  File "c:\Users\Delta\YOLOv8_GUtech_Turtle\Turtle_Detection\Detect_Track.py", line 481, in videoTrack
    tracker_objects = tracker.update(numpy_detections, frame)
  File "c:\users\delta\yolov8_gutech_turtle\turtle_detection\yolo_tracking\boxmot\trackers\bytetrack\byte_tracker.py", line 217, in update
    STrack.multi_predict(strack_pool)
  File "c:\users\delta\yolov8_gutech_turtle\turtle_detection\yolo_tracking\boxmot\trackers\bytetrack\byte_tracker.py", line 38, in multi_predict
    multi_mean, multi_covariance = STrack.shared_kalman.multi_predict(multi_mean, multi_covariance)
  File "c:\users\delta\yolov8_gutech_turtle\turtle_detection\yolo_tracking\boxmot\motion\adapters\bytetrack_kf_adapter.py", line 157, in multi_predict
    for i in range(len(x)):
NameError: name 'x' is not defined
```

The first frame is tracked, but on the second frame, this issue occurs (the tracking algorithm works on individual frames alone though but not as part of the video). 
I'm sorry but am I using this tracker in the wrong manner?

**Comment by mikel-brostrom** [comment]
@henriksod Maybe the adapters are not working as intended?

**Comment by ElNoSabe322** [comment]
okay lol 😆 the error is appearing anymore. 
Did you manage to fix it? Regardless, thank you so much @mikel-brostrom @henriksod

**Comment by mikel-brostrom** [comment]
Yup, we rolled back the adapters. Went back to individual KF again.

**Comment by saurabheights** [comment]
@mikel-brostrom 
> Yup, we rolled back the adapters. Went back to individual KF again.

Just FYI, probably these changes have caused boxmot to break for version 10.0.17, works fine in 10.0.16. Importing Bytetracker fails now on latest pypi version.

```
  File "/home/sk/workspace/multitrackpro/multitracker/core/tracker/multi_tracker.py", line 5, in <module>
    from boxmot import BYTETracker
  File "/home/sk/miniconda3/envs/MultiTrackPro/lib/python3.10/site-packages/boxmot/__init__.py", line 4, in <module>
    from boxmot.trackers.botsort.bot_sort import BoTSORT
  File "/home/sk/miniconda3/envs/MultiTrackPro/lib/python3.10/site-packages/boxmot/trackers/botsort/bot_sort.py", line 8, in <module>
    from boxmot.motion.kalman_filters.adapters import BotSortKalmanFilterAdapter
ModuleNotFoundError: No module named 'boxmot.motion.kalman_filters'
```

Attached screenshot of source code from pypi source. 

![Screenshot from 2023-07-27 14-18-20](https://github.com/mikel-brostrom/yolo_tracking/assets/11634769/9f575c61-58de-4914-a141-85e59a8173dd)
![Screenshot from 2023-07-27 14-14-58](https://github.com/mikel-brostrom/yolo_tracking/assets/11634769/ba8594d4-cea4-416f-98d3-b9d230746a7a)

**Comment by mikel-brostrom** [comment]
I have no issues with this using `10.0.17`. Moreover this is covered by our tests here:

https://github.com/mikel-brostrom/yolo_tracking/blob/b0a92b3d8e2b07d4c8844e7ce3e97e1a6dc71eae/tests/test_python.py#L44

**Comment by maniklem** [comment]
> 

I am having the exact same error after upgrading to version 10.0.17. 

`Traceback (most recent call last):
  File "C:\Users\PC\Documents\software\yolo_tracking-130723\examples\YOLO_tracking.py", line 7, in <module>
    from boxmot import DeepOCSORT
  File "C:\Users\PC\anaconda3\lib\site-packages\boxmot\__init__.py", line 4, in <module>
    from boxmot.trackers.botsort.bot_sort import BoTSORT
  File "C:\Users\PC\anaconda3\lib\site-packages\boxmot\trackers\botsort\bot_sort.py", line 8, in <module>
    from boxmot.motion.kalman_filters.adapters.botsort_kf_adapter import BotSortKalmanFilterAdapter
ModuleNotFoundError: No module named 'boxmot.motion.kalman_filters'`

**Comment by henriksod** [comment]
Hi, it seems there is an __init__.py file missing from the boxmot/motion directory. This is probably the cause of this issue.

**Comment by mikel-brostrom** [comment]
https://github.com/mikel-brostrom/yolo_tracking/blob/master/boxmot/motion/__init__.py

**Comment by mikel-brostrom** [comment]
I tried this on a new environment, no issues.

**Comment by ElNoSabe322** [comment]
> I tried this on a new environment, no issues.

Same. ByteTrack works fine for me as well.

**Comment by mikel-brostrom** [comment]
Thx for letting us know @ElNoSabe322  :smile:

**Comment by mikel-brostrom** [comment]
I don't know where I was looking. As @henriksod pointed out there was a missing `__init__.py`. This is now fixed in `10.0.18`

https://github.com/mikel-brostrom/yolo_tracking/issues/1036
